src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader_search.cs

```diff
@@ -90,6 +131,76 @@ public bool TryReadToInternal(out ReadOnlySequence<T> sequence, T delimiter, boo
             return false;
         }
 
+        /// <summary>
+        /// Try to read everything up to the given <paramref name="delimiter"/>, ignoring delimiters that are
+        /// preceeded by <paramref name="skipDelimiter"/>.
+        /// </summary>
+        /// <param name="sequence">The read data, if any.</param>
+        /// <param name="delimiter">The delimiter to look for.</param>
+        /// <param name="skipDelimiter">If found prior to <paramref name="delimiter"/> it will skip that occurence.</param>
+        /// <param name="advancePastDelimiter">True to move past the <paramref name="delimiter"/> if found.</param>
+        /// <returns>True if the <paramref name="delimiter"/> was found.</returns>
+        public bool TryReadTo(out ReadOnlySequence<T> sequence, T delimiter, T skipDelimiter, bool advancePastDelimiter = true)
+        {
+            return TryReadToInternal(out sequence, delimiter, skipDelimiter, advancePastDelimiter);
+        }
+
+        public bool TryReadToInternal(out ReadOnlySequence<T> sequence, T delimiter, T skipDelimiter, bool advancePastDelimiter, int skip = 0)
+        {
+            BufferReader<T> copy = this;
+            if (skip > 0)
+            {
+                if (skip > 1)
+                {
+                    Advance(skip - 1);
+                }
+
+                TryRead(out T prior);
+                if (prior.Equals(skipDelimiter) && !End)
+                {
+                    Advance(1);
+                }
+            }
+
+            ReadOnlySpan<T> remaining = UnreadSpan;
+            bool shouldSkipDelimiter = false;
+
+            while (!End)
+            {
+                int index = remaining.IndexOf(delimiter);
+                if (index != -1)
+                {
+                    if ((shouldSkipDelimiter && index == 0) || (index > 0 && remaining[index - 1].Equals(skipDelimiter)))
+                    {
+                        // This delimiter should be skipped, move past it.
+                        shouldSkipDelimiter = false;
+                        Advance(index + 1);
+                        remaining = UnreadSpan;
+                        continue;
+                    }
+
+                    // Found the delimiter. Move to it, slice, then move past it.
+                    sequence = Sequence.Slice(copy.Position, Position);
```

Comment:
What happens if the skipDelimiter character itself has been escaped?
`abc\\"def"` should return `abc\\`

That is why I had to keep track of number of `skipDelimiter` characters seen and mod that by 2:
https://github.com/dotnet/corefxlab/blob/ad31432950ab70b3ddb51ac982d36bb56da64f1f/src/System.Text.JsonLab/System/Text/Json/Utf8JsonReader_MultiSegment.cs#L385

This is a decent test case that I used:
https://github.com/dotnet/corefxlab/blob/ad31432950ab70b3ddb51ac982d36bb56da64f1f/tests/System.Text.JsonLab.Tests/JsonReaderTests.cs#L295

Reply:
I'll add the tests and update if needed.

Reply:
I've got the fix for multiple escape bytes- took awhile to get it down as there was a bug in our test implementation of `ReadOnlySequenceSegment<byte>` that I accidentally stumbled over.